jupyterhub/templates/hub/regsecret.yaml

```diff
@@ -0,0 +1,10 @@
+{{- if .Values.singleuser.image.createPullSecret }}
+kind: Secret
+apiVersion: v1
+metadata:
+  name: {{ .Values.singleuser.image.pullSecret.name }}
+  namespace: {{ .Release.Namespace }}
```

Comment:
- We don't have to set the namespace explicitly.
- The file itself should be moved to the singleuser folder.
- We should add labels to it:
```yaml
metadata:
  name: singleuser-image-credentials
  labels:
    {{- $_ := merge (dict "componentSuffix" "-image-credentials") . }}
    {{- include "jupyterhub.labels" $_ | nindent 4 }}
```

Reply:
You could name the file `image-credentials-secret.yaml` perhaps.

Reply:
I will move the file it makes more sense to put it in there. 

I am pretty sure this secret needs to be in the same namespace as the pods being spun up. dockerconfig secrets are not global.

Reply:
The namespace is decided implicitly, if you look at other files, they all end up in the correct namespace even though it isn't explicitly written out. This would work as well though!

Reply:
oh that makes sense.